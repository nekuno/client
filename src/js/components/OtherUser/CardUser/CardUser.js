import PropTypes from 'prop-types';
import React, { Component } from 'react';
import styles from './CardUser.scss';
import RoundedImage from '../../ui/RoundedImage/RoundedImage.js';
import ProgressBar from '../../ui/ProgressBar/ProgressBar.js';
import translate from '../../../i18n/Translate';

@translate('CardUser')
export default class CardUser extends Component {

    static propTypes = {
        photo         : PropTypes.string.isRequired,
        nickname      : PropTypes.string.isRequired,
        age           : PropTypes.number.isRequired,
        city          : PropTypes.string.isRequired,
        matching      : PropTypes.number.isRequired,
        similarity    : PropTypes.number.isRequired,
        coincidences  : PropTypes.number.isRequired,
        group         : PropTypes.object,
        size          : PropTypes.oneOf(['small', 'medium']).isRequired,
        onClickHandler: PropTypes.func
    };

    handleClick() {
        if (this.props.onClickHandler) {
            this.props.onClickHandler();
        }
    }

    render() {
        const {photo, nickname, age, city, matching, similarity, coincidences, group, size, strings} = this.props;

        return (
            <div className={styles.cardUser + ' ' + styles[size]} onClick={this.handleClick.bind(this)}>
                <div className={styles.frame}>
                    {group && group.name ?
                        <div className={styles.group}>
                            <RoundedImage url={group.photo} size="xx-small"/>
                            <div className={styles.groupName}>{group.name}</div>
                        </div>
                        : null
                    }
                    <div className={styles.userImage}>
                        <img src={photo}/>
                    </div>
                    <div className={styles.topData}>
                        <div className={styles.nickname}>{nickname}</div>
                        <div className={styles.ageCity}>{city} &bull; {age}</div>
                    </div>
                    <div className={styles.progressBars}>
                        <div className={styles.progressBarTitle}>{strings.compatible}&nbsp;</div>
                        <div className={styles.progressBar}>
                            <ProgressBar percentage={matching} size={'small'} strokeColor={'#756EE5'}/>
                        </div>
                        <div className={styles.progressBarTitle}>{strings.similar}&nbsp;</div>
                        <div className={styles.progressBar}>
                            <ProgressBar percentage={similarity} size={'small'} strokeColor={'#756EE5'}/>
                        </div>
                    </div>
                    <div className={styles.coincidences}>{coincidences} {strings.coincidences}</div>
                </div>
            </div>
        );
    }
}

CardUser.defaultProps = {
    strings: {
        compatible  : 'Compatible',
        similar     : 'Similar',
        coincidences: 'Coincidences',
    }
};