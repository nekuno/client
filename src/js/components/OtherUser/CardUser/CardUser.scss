.card-user {
  display: inline-block;
  font-weight: bold;
  margin: 0;
  width: 50%;

  .user-image {
    text-align: center;
  }
  .top-data {
    display: inline-block;
    text-align: center;
    width: 100%;
    .nickname {
      font-size: 20px;
    }
    .age-city {
      font-size: 14px;
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
    }
    .gender {
      font-size: 14px;
      margin: 15px 0;
      text-align: left;
    }
  }
  .coincidences {
    margin: 10px 0;
    font-size: 14px;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }
  .networks {
    height: 38px;
    overflow: hidden;
  }
  .resume {
    display: none;
    font-size: 13px;
    font-weight: normal;
    margin: 10px 0;
  }

  &.medium {
    width: 75%;

    .user-image {
      display: inline-block;
      text-align: left;
    }
    .top-data {
      display: inline-block;
      margin: 20px 10px;
      text-align: left;
      vertical-align: top;
      width: 30%;

      .nickname {
      }
      .age-city {
        margin: 5px 0;
      }
      .gender {
        margin: 5px 0;
      }
    }
    .coincidences {
      display: inline-block;
      margin-top: 15px;
    }
    .networks {
      display: inline-block;
    }
    .resume {
      display: block;
    }
  }
}