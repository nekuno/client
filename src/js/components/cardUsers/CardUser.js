import PropTypes from 'prop-types';
import React, { Component } from 'react';
import ProgressBar from '../ui/ProgressBar';
import Button from '../ui/Button';
import Image from '../ui/Image';
import CardUserTopLinks from '../recommendations/CardUserTopLinks';
import * as UserActionCreators from '../../actions/UserActionCreators'
import translate from '../../i18n/Translate';
import PercentageValue from "./PercentageValue";
import Framework7Service from "../../services/Framework7Service";

@translate('CardUser')
export default class CardUser extends Component {

    static contextTypes = {
        router: PropTypes.object.isRequired
    };

    static propTypes = {
        userId                : PropTypes.number.isRequired,
        username              : PropTypes.string.isRequired,
        slug                  : PropTypes.string.isRequired,
        location              : PropTypes.string,
        canSendMessage        : PropTypes.bool.isRequired,
        photo                 : PropTypes.object,
        matching              : PropTypes.number.isRequired,
        similarity            : PropTypes.number,
        age                   : PropTypes.number,
        like                  : PropTypes.number,
        hideLikeButton        : PropTypes.bool.isRequired,
        loggedUserSlug        : PropTypes.string.isRequired,
        profile               : PropTypes.object.isRequired,
        handleSelectProfile   : PropTypes.func,
        online                : PropTypes.bool,
        topLinks              : PropTypes.array,
        sharedLinks           : PropTypes.number,
        orientationMustBeAsked: PropTypes.bool,

        // Injected by @translate:
        strings       : PropTypes.object
    };

    constructor(props) {
        super(props);

        this.onLikeOrDislike = this.onLikeOrDislike.bind(this);
        this.handleMessage = this.handleMessage.bind(this);
        this.handleGoToProfile = this.handleGoToProfile.bind(this);
    }

    onLikeOrDislike() {
        const {like, loggedUserSlug, slug} = this.props;
        if (!like || like === -1) {
            UserActionCreators.likeUser(loggedUserSlug, slug);
        } else {
            UserActionCreators.deleteLikeUser(loggedUserSlug, slug);
        }
    }

    handleMessage() {
        this.context.router.push(`/conversations/${this.props.slug}`);
    }

    handleGoToProfile() {
        const {orientationMustBeAsked, slug} = this.props;
        if (orientationMustBeAsked) {
            Framework7Service.nekunoApp().popup('.popup-orientation-required');
            this.props.handleSelectProfile(slug);
        } else {
            this.context.router.push(`/p/${slug}`);
        }
    }

    render() {
        const {canSendMessage, like, hideLikeButton, photo, userId, username, matching, similarity, age, online, topLinks, sharedLinks, strings} = this.props;
        const location = this.props.location || strings.noLocation;
        const subTitle = location.substr(0, 15) + (location.length > 15 ? '...' : '') + ' Â· ' + age;
        const messageButton = canSendMessage ? <span className="icon-message" onClick={this.handleMessage}></span> : '';
        const likeButtonText = like === null ? strings.saving : like ? strings.unlike : strings.like;
        const likeButton = hideLikeButton ? '' : <div className="like-button-container"><Button onClick={this.onLikeOrDislike} disabled={like === null ? 'disabled' : null}>{likeButtonText}</Button></div>;
        const defaultSrc = 'img/no-img/big.jpg';
        let imgSrc = photo ? photo.thumbnail.big : defaultSrc;

        return (
            <div className="card person-card" onClick={this.handleGoToProfile}>
                <div className="card-header">
                    <div className="card-content">
                        <div className="card-content-inner">
                            <div className="image fixed-max-height-image">
                                <Image src={imgSrc} defaultSrc={defaultSrc}/>
                            </div>
                        </div>
                        <div className="card-content-overlay">
                            {like ?
                                <div className="like-icon-container"><span className="icon-star"></span></div>
                                : null
                            }
                            <div className="data-container">
                                <div className="card-title">{username}</div>
                                <div className="card-sub-title">{subTitle}</div>
                                {/* {online ? <div className="online-status">Online</div> : null}
                                <div className="send-message-button icon-wrapper">
                                {messageButton}
                                </div> */}
                            </div>
                        </div>
                    </div>
                </div>
                {/* <CardUserTopLinks topLinks={topLinks} sharedLinks={sharedLinks} onClick={this.handleGoToProfile}/> */}
                <div className={"card-footer"}>
                    <div>
                        {/* <div className="card-title">
                            {username}
                        </div> */}

                        <div className="matching-string">{strings.matching}</div>
                        <div className="matching-value">
                            <ProgressBar percentage={matching}/>
                            <div className="matching-percentage">{(matching ? matching : 0) + ' %'}</div>
                        </div>
                        
                        <div className="matching-string">{strings.similarity}</div>
                        <div className="matching-value">
                            <ProgressBar percentage={similarity}/>
                            <div className="matching-percentage">{(similarity ? similarity : 0) + ' %'}</div>
                        </div>
                    </div>
                </div>
            </div>
        );
    }

}

CardUser.defaultProps = {
    strings: {
        like      : 'Like',
        unlike    : 'Remove',
        matching  : 'Matching',
        similarity: 'Similarity',
        saving    : 'Saving...',
        age       : 'Age',
    }
};