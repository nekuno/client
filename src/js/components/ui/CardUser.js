import React, { PropTypes, Component } from 'react';
import ProgressBar from './ProgressBar';
import Button from './Button';
import Image from './Image';
import * as UserActionCreators from '../../actions/UserActionCreators'
import translate from '../../i18n/Translate';

@translate('CardUser')
export default class CardUser extends Component {

    static contextTypes = {
        history: PropTypes.object.isRequired
    };

    static propTypes = {
        userId        : PropTypes.number.isRequired,
        username      : PropTypes.string.isRequired,
        location      : PropTypes.string,
        canSendMessage: PropTypes.bool.isRequired,
        photo         : PropTypes.object,
        matching      : PropTypes.number.isRequired,
        age           : PropTypes.number,
        like          : PropTypes.number,
        hideLikeButton: PropTypes.bool.isRequired,
        loggedUserId  : PropTypes.number.isRequired,
        profile       : PropTypes.object.isRequired,
        handleSelectProfile: PropTypes.func,

        // Injected by @translate:
        strings       : PropTypes.object
    };

    constructor(props) {
        super(props);

        this.onLikeOrDislike = this.onLikeOrDislike.bind(this);
        this.handleMessage = this.handleMessage.bind(this);
        this.handleGoToProfile = this.handleGoToProfile.bind(this);
    }

    onLikeOrDislike() {
        const {like, loggedUserId, userId} = this.props;
        if (!like || like === -1) {
            UserActionCreators.likeUser(loggedUserId, userId);
        } else {
            UserActionCreators.deleteLikeUser(loggedUserId, userId);
        }
    }

    handleMessage() {
        this.context.history.pushState(null, `/conversations/${this.props.userId}`);
    }

    handleGoToProfile() {
        const {userId, profile} = this.props;
        if (!profile.orientation) {
            nekunoApp.popup('.popup-orientation-required');
            this.props.handleSelectProfile(userId);
        } else {
            this.context.history.pushState(null, `/profile/${userId}`);
        }
    }

    render() {
        const {location, canSendMessage, like, hideLikeButton, photo, userId, username, matching, age, strings} = this.props;
        const subTitle = <div><span className="icon-marker"></span>{location.substr(0, 15)}{location.length > 15 ? '...' : ''} - {strings.age}: {age}</div>;
        const messageButton = canSendMessage ? <span className="icon-message" onClick={this.handleMessage}></span> : '';
        const likeButtonText = like === null ? strings.saving : like ? strings.unlike : strings.like;
        const likeButton = hideLikeButton ? '' : <div className="like-button-container"><Button onClick={this.onLikeOrDislike} disabled={like === null ? 'disabled' : null}>{likeButtonText}</Button></div>;
        const defaultSrc = 'img/no-img/big.jpg';
        let imgSrc = photo ? photo.thumbnail.big : defaultSrc;

        return (
            <div className="card person-card">
                <div className="card-header">
                    <div className="card-title" onClick={this.handleGoToProfile}>
                        {username}
                    </div>
                    <div className="card-sub-title">
                        {subTitle}
                    </div>
                    {/*<div className="send-message-button icon-wrapper">*/}
                    {/*{messageButton}*/}
                    {/*</div>*/}
                </div>
                <div className="card-content">
                    <div className="card-content-inner">
                        <div className="image fixed-height-image" onClick={this.handleGoToProfile}>
                            <Image src={imgSrc} defaultSrc={defaultSrc}/>
                        </div>
                        {like ? <div className="like-icon-container" onClick={this.handleGoToProfile}><span className="icon-star"></span></div> : null}
                        <div className="matching">
                            <ProgressBar percentage={matching}/>
                        </div>
                    </div>
                </div>
                <div className="card-footer">
                    <div className="matching-value">{strings.matching} {matching ? matching + '%' : '0%'}</div>
                </div>
            </div>
        );
    }

}

CardUser.defaultProps = {
    strings: {
        like    : 'Like',
        unlike  : 'Remove',
        matching: 'Matching',
        saving  : 'Saving...',
        age     : 'Age',
    }
};