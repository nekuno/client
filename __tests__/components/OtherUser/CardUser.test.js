import React from 'react';
import CardUser from '../../../src/js/components/OtherUser/CardUser/CardUser.js';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Test CardUser component', () => {
    const mockFn = jest.fn();
    const cardUser1 = shallow(
        <CardUser onClickHandler={mockFn}
                  photo={'http://via.placeholder.com/100x100'}
                  nickname={'JohnDoe'}
                  age={36}
                  city={'New York'}
                  gender={'Male'}
                  matching={76}
                  similarity={85}
                  coincidences={24}
                  networks={['facebook', 'twitter', 'steam']}
                  size="small"
                  resume="foo"
        />
    );
    const cardUser2 = shallow(
        <CardUser onClickHandler={mockFn}
                  photo={'http://via.placeholder.com/100x100'}
                  nickname={'JaneDoe'}
                  age={30}
                  city={'New Jersey'}
                  gender={'Female'}
                  matching={89}
                  similarity={34}
                  coincidences={101}
                  networks={['tumblr', 'spotify', 'instagram', 'facebook', 'twitter', 'steam']}
                  size="medium"
                  resume="bar"
        />
    );

    it('should be defined', () => {
        expect(CardUser).toBeDefined();
    });
    it('should render correctly', () => {
        expect(cardUser1).toMatchSnapshot();
        expect(cardUser2).toMatchSnapshot();
    });
});
