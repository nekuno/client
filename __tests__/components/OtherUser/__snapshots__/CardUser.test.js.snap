// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CardUser component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TranslationComponent
    age={36}
    city="New York"
    coincidences={24}
    gender="Male"
    matching={76}
    networks={
      Array [
        "facebook",
        "twitter",
        "steam",
      ]
    }
    nickname="JohnDoe"
    onClickHandler={[MockFunction]}
    photo="http://via.placeholder.com/100x100"
    resume="foo"
    similarity={85}
    size="small"
    strings={
      Object {
        "coincidences": "Coincidences",
        "matching": "Matching",
        "similarity": "Similarity",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "age": 36,
      "city": "New York",
      "coincidences": 24,
      "gender": "Male",
      "locale": undefined,
      "matching": 76,
      "networks": Array [
        "facebook",
        "twitter",
        "steam",
      ],
      "nickname": "JohnDoe",
      "onClickHandler": [MockFunction],
      "photo": "http://via.placeholder.com/100x100",
      "resume": "foo",
      "similarity": 85,
      "size": "small",
      "strings": Object {
        "coincidences": "Coincidences",
        "matching": "Matching",
        "similarity": "Similarity",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "age": 36,
        "city": "New York",
        "coincidences": 24,
        "gender": "Male",
        "locale": undefined,
        "matching": 76,
        "networks": Array [
          "facebook",
          "twitter",
          "steam",
        ],
        "nickname": "JohnDoe",
        "onClickHandler": [MockFunction],
        "photo": "http://via.placeholder.com/100x100",
        "resume": "foo",
        "similarity": 85,
        "size": "small",
        "strings": Object {
          "coincidences": "Coincidences",
          "matching": "Matching",
          "similarity": "Similarity",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Test CardUser component should render correctly 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TranslationComponent
    age={30}
    city="New Jersey"
    coincidences={101}
    gender="Female"
    matching={89}
    networks={
      Array [
        "tumblr",
        "spotify",
        "instagram",
        "facebook",
        "twitter",
        "steam",
      ]
    }
    nickname="JaneDoe"
    onClickHandler={[MockFunction]}
    photo="http://via.placeholder.com/100x100"
    resume="bar"
    similarity={34}
    size="medium"
    strings={
      Object {
        "coincidences": "Coincidences",
        "matching": "Matching",
        "similarity": "Similarity",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "age": 30,
      "city": "New Jersey",
      "coincidences": 101,
      "gender": "Female",
      "locale": undefined,
      "matching": 89,
      "networks": Array [
        "tumblr",
        "spotify",
        "instagram",
        "facebook",
        "twitter",
        "steam",
      ],
      "nickname": "JaneDoe",
      "onClickHandler": [MockFunction],
      "photo": "http://via.placeholder.com/100x100",
      "resume": "bar",
      "similarity": 34,
      "size": "medium",
      "strings": Object {
        "coincidences": "Coincidences",
        "matching": "Matching",
        "similarity": "Similarity",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "age": 30,
        "city": "New Jersey",
        "coincidences": 101,
        "gender": "Female",
        "locale": undefined,
        "matching": 89,
        "networks": Array [
          "tumblr",
          "spotify",
          "instagram",
          "facebook",
          "twitter",
          "steam",
        ],
        "nickname": "JaneDoe",
        "onClickHandler": [MockFunction],
        "photo": "http://via.placeholder.com/100x100",
        "resume": "bar",
        "similarity": 34,
        "size": "medium",
        "strings": Object {
          "coincidences": "Coincidences",
          "matching": "Matching",
          "similarity": "Similarity",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
